### PR TITLE
feat: add a way to override kubelet configuration via machine config

### LINKED ISSUE #4629: Kubelet configuration available via machine configuration
Kubelet command line flags are getting deprecated, and new way to use the config file.

Talos today accepts only `extraArgs`, so we need a better way to fine-tune kubelet.

We don't want to import whole `KubeletConfig` as a Go structure into the machine configuration as this will cause unnecessary dependency. We would rather present `KubeletConfig` as a `map[string]interface{}`, we skip any validation on Talos side, and simply unmarshal raw data into the `KubeletConfig` before starting the `kubelet`. Kubelet will validate its config and error out if it's invalid.

### PR DESCRIPTION
**Pull request by smira**
Fixes #4629

Note: some fields are enforced by Talos and are not overridable.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/5045)

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go [hunk 1]
```diff
@@ -191,49 +195,80 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
...
+	for field, message := range map[string]string{
+		"staticPodPath":  "staticPodPath can't be overridden",
+		"authentication": "authentication can't be overridden",
+		"authorization":  "authorization can't be overridden",
+		"cgroupRoot":     "cgroup configuration can't be overridden",
+		"systemCgroups":  "cgroup configuration can't be overridden",
+		"kubeletCgroups": "cgroup configuration can't be overridden",
+		"port":           "port can't be overridden",
+		"apiVersion":     "apiVersion can't be overridden",
+		"kind":           "kind can't be overridden",
```
Comment (JAORMX) [inline internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:212]: Should these be documented?

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:212]: yes, they should.

this is a work in progress PR, it's not finished yet

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec_test.go [hunk 2]
```diff
@@ -184,3 +241,102 @@ func (suite *KubeletSpecSuite) TearDownTest() {
...
+				"Port": "xyz",
+			},
+			expectedErr: "error unmarshalling extra kubelet configuration: strict decoding error: unknown field \"API\", unknown field \"Port\", unknown field \"foo\"",
+		},
+		{
+			name: "wrong field type",
+			extraConfig: map[string]interface{}{
+				"oomScoreAdj": "v1",
+			},
+			expectedErr: "error unmarshalling extra kubelet configuration: unrecognized type: int32",
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/kubelet_spec_test.go:265]: shouldn't it say `string`  instead?

Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec_test.go:265]: it's not us in this case, it's the way Kubernetes converter unmarshaller the error, and we have no control over it

Comment (frezbo) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec_test.go:265]: okay, it was confusing though :sweat_smile:

**Review by Ulexus — APPROVED** [review]
> Reviewed 6 of 11 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/talos-systems/talos/5045)* status: all files reviewed, 2 unresolved discussions (waiting on @JAORMX and @smira)

**Review by frezbo — APPROVED** [review]

**Pull request merged**